kernel/include/timeout_q.h

```diff
@@ -22,13 +22,18 @@ extern "C" {
 
 static inline void _init_timeout(struct _timeout *t, _timeout_func_t fn)
 {
-	t->dticks = _INACTIVE;
+	sys_dnode_init(&t->node);
 }
 
 void _add_timeout(struct _timeout *to, _timeout_func_t fn, s32_t ticks);
 
 int _abort_timeout(struct _timeout *to);
 
+static inline bool _is_inactive_timeout(struct _timeout *t)
```

Comment:
Not particularly relevant, but this was always a bad API (a true return indicates a false state) that never got fixed up during all the refactoring.  I wouldn't cry to see this patch change it to z_timeout_active() or whatever.

Reply:
This was partly because of the legacy of `_EXPIRED`: timeouts could be in three states, not two.  This function tests that the state is exactly what `_INACTIVE` indicated.

A true result indicates that the predicate identified by the function name is true.  Why is that a bad API?  We use `sys_dlist_is_empty()`, not `sys_dlist_is_nonempty()`.

The whole timeout API uses prefix underscore.  While I'd rather that wasn't the case (violates C reserved identifiers for the non-static functions) I wouldn't want to change just one function, nor change all of them, in this PR.

Reply:
> A true result indicates that the predicate identified by the function name is true

I don't think that not incorrect, exactly, but it's not innapropriate to argue against the ellision of unommitted negatives either.

No seriously, double negatives in API choices are bad, especially in predicates.  The base adjective for that state is "active".  True should mean "active", false should mean "inactive".